sphinx/themes/basic/static/searchtools.js

```diff
@@ -219,7 +219,14 @@ const Search = {
     (document.body.appendChild(document.createElement("script")).src = url),
 
   setIndex: (index) => {
-    Search._index = index;
+    const sanitiseRecursive = function(obj) {
+        if (obj instanceof Object && !Array.isArray(obj)) {
+            Object.entries(obj).map(([k, v]) => obj[k] = sanitiseRecursive(v));
+            return Object.freeze(Object.assign(Object.create(null), obj));
+        }
+        return Object.freeze(obj);
+    }
+    Search._index = sanitiseRecursive(index);
```

Comment:
Question: what is the client-side performance impact of this on load-time for large Sphinx HTML documentation projects?

Reply:
> Question: what is the client-side performance impact of this on load-time for large Sphinx HTML documentation projects?

The answer to this seems to be: too much additional overhead to be worthwhile, given that there is no known vulnerability.  A `sanitiseRecursive` on the Python3.13 documentation `searchindex.js` index content adds ~160ms to the search results page when I attempt it locally from 58ca7000dc2bbaeda556c74ebb8b5b74e6b62325.

![image](https://github.com/user-attachments/assets/5ff8c6ed-614e-4d3b-9482-87df0a853740)

Comparatively on a baseline built against 7801bd77b8ebf0bddfedc00f0d61cea56f3f4e2f, the calling `setIndex` function completes so quickly that the profiler does not capture any traces from it:

![image](https://github.com/user-attachments/assets/78d3ad61-104d-4371-9146-07315fc9598a)

Reply:
With performance refactorings up to 6a38587e73a9ce686e21008cdad228c10f34b33d, that's reduced to ~75ms overhead.  If possible I'd like to achieve <= 50ms.

![image](https://github.com/user-attachments/assets/d69a02a9-e8e5-4a6c-933f-9021cd79b2b9)

Reply:
Closing this thread as outdated; performance costs necessarily increased after these benchmarks, when it was discovered that [nested arrays were not being frozen](https://github.com/sphinx-doc/sphinx/pull/13099#discussion_r1829149466).